src/black/strings.py

```diff
@@ -236,3 +237,40 @@ def normalize_string_quotes(s: str) -> str:
         return s  # Prefer double quotes
 
     return f"{prefix}{new_quote}{new_body}{new_quote}"
+
+
+def char_width(char: str) -> int:
+    """Return the width of a single character as it would be displayed in a
+    terminal or editor (which respects Unicode East Asian Width).
+
+    Full width characters are counted as 2, while half width characters are
+    counted as 1.
+    """
+    return 2 if unicodedata.east_asian_width(char) in ("F", "W") else 1
+
+
+def str_width(line_str: str) -> int:
+    """Return the width of `line_str` as it would be displayed in a terminal
+    or editor (which respects Unicode East Asian Width).
+
+    You could utilize this function to determine, for example, if a string
+    is too wide to display in a terminal or editor.
+    """
+    if line_str.isalpha():
```

Comment:
```suggestion
    if line_str.isascii():
```
`isascii` is O(1) in CPython because string objects store internally whether they're ASCII-only. `isalpha` has to actually loop over the string. It's also not obvious that `isalpha` is correct; I'd have to look up whether any East Asian characters are counted as "alpha" for Unicode purposes.

Reply:
Actually, it was not intended, but just a typo.  I amended the commit!

Reply:
strictly speaking `isascii` does not ensure a string width being the same as its length since `\u0000`~`\u001f` are control characters with no width. But that should not be a problem for formatting code